thingsvision/utils/models/mae/vit_mae.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+# --------------------------------------------------------
+# References:
+# timm: https://github.com/rwightman/pytorch-image-models/tree/master/timm
+# DeiT: https://github.com/facebookresearch/deit
+# --------------------------------------------------------
+
+from functools import partial
+
+import torch
+import torch.nn as nn
+from timm.models import vision_transformer
+
+
+class VisionTransformer(vision_transformer.VisionTransformer):
+    """Vision Transformer with support for global average pooling"""
+
+    def __init__(self, **kwargs):
+        super(VisionTransformer, self).__init__(**kwargs)
+
+    def forward_features(self, x):
+        B = x.shape[0]
+        x = self.patch_embed(x)
+
+        cls_tokens = self.cls_token.expand(
+            B, -1, -1
+        )  # stole cls_tokens impl from Phil Wang, thanks
+        x = torch.cat((cls_tokens, x), dim=1)
```

Comment:
In [deit/models_v2.py]( https://github.com/facebookresearch/deit/blob/main/models_v2.py ) it is 
```
        cls_tokens = self.cls_token.expand(B, -1, -1)
        
        x = x + self.pos_embed
        
        x = torch.cat((cls_tokens, x), dim=1)
```
i.e., concatenation after the addition of positional embedding, while here, the other way around.

Reply:
I mainly copied from the official MAE repo. See [here](https://github.com/facebookresearch/mae/blob/main/models_vit.py#L38-L40).

Reply:
Ok, that makes sense. I just looked up the links referenced at the top of the `vit_mae.py` file.